crowd_anki/config/config_settings.py

```diff
@@ -0,0 +1,74 @@
+from enum import Enum
+from dataclasses import dataclass
+
+from aqt import mw
+from ..utils.constants import USER_FILES_PATH
+
+@dataclass
+class ConfigSettings():
+    class Properties(Enum):
+        SNAPSHOT_PATH = "snapshot_path"
+        AUTOMATED_SNAPSHOT = "automated_snapshot"
+        SNAPSHOT_ROOT_DECKS = "snapshot_root_decks"
+        EXPORT_DECK_SORT_METHODS = "export_deck_sort_method"
+        EXPORT_DECK_SORT_REVERSED = "export_deck_sort_reversed"
+
+
+    class DeckExportSortMethods(Enum):
+        NO_SORTING = "none"
+        GUID = "guid"
+        FLAG = "flag"
+        TAG = "tag"
+        NOTEMODEL = "notemodel"
+        NOTEMODELID = "notemodelid"
+        FIELD1 = "field1"
+        FIELD2 = "field2"
+
+    snapshot_path: str
+    automated_snapshot: bool
+    snapshot_root_decks: list
+
+    export_deck_sort_reversed: bool
+    export_deck_sort_methods: list
+
+
+    def __init__(self, get_settings = True):
+        if get_settings:
+            self.config_file = mw.addonManager.getConfig(__name__)
+            self.get_all_values()
+
+    def get_all_values(self):
+        self.snapshot_path = self.config_file.get(self.Properties.SNAPSHOT_PATH.value, str(USER_FILES_PATH.resolve()))
```

Comment:
I think it'd better if the default values would be explicitly defined in the property enum. So do something like:

```
ConfigEntry = namedtuple("ConfigEntry", ["name", "default_value"])

class Properties(Enum):
        SNAPSHOT_PATH = ConfigEntry("snapshot_path", str(USER_FILES_PATH.resolve()))
...
```

Reply:
Also this would benefit from a bit of metaprogramming. Something like:

```
    def _get(self, prop: Properties):
        self.config_file.get(prop.value.config_name, prop.value.default_value)

    def load_values(self):
        for prop in self.Properties:
            setattr(self, prop.value, self._get(prop))
```

Reply:
hmm, you don't need to use getattr to access them in this case. As a minimal scenario something like this works:

```python
class Dummy:
    pass

x = Dummy()

setattr(x, "name", "value")

print(x.name)
```

with type hints you can make it more obvious what values your class can vend + keep autocompletion working, so you can do something like this:

```python
@dataclass
class Dummy2:
    name: str = field(init=False)

y = Dummy2()

setattr(y, "name", "value2")

print(y.name)
```

Reply:
Ah I see, I believe it wasn't pep8 though, but Pycharm complaining that it can't figure out what those values are (which as you mentioned is fair 😛). The approach 2 above should remediate that though.

Reply:
Is there any benefit to changing the current class to a dataclass with nothing initialised? There are already no warnings :wink: 

```
class ConfigSettings:
    snapshot_path: str
    automated_snapshot: bool
    snapshot_root_decks: list
    export_notes_reverse_order: bool
    export_note_sort_methods: list

```

Reply:
Nope, given that we don't use the dataclass functionality it's actually better like it is right now - just with type hints :)